### PR TITLE
Add basic debug tools (plots)

### PR DESCRIPTION
**Pull request by amoodie**
Add some basic debugging plots.

These are helpful for exploring where weird index errors and status bugs arise, with the help of the Python Debugging console.

Add tests, and documentation for the debugging tools.

Example:

```python
self.show_attribute('cell_type')
self.show_ind([15, 16, 17, 19, 20])
plt.show()
```
image

### CODE REVIEW

File: pyDeltaRCM/debug_tools.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+        Private method called by :obj:`show_attribute`.
+        """
+        if not ax:
+            ax = plt.gca()
+
+        ax.imshow(getattr(self, attribute),
+                  cmap=plt.get_cmap('viridis'),
+                  interpolation='none')
+        ax.autoscale(False)
```
Comment (elbeejay) [inline pyDeltaRCM/debug_tools.py]: Can the colorbar be added here? Would be helpful to know when looking at the cell type plot for example, which cell types correspond to which colors on the plot.

File: pyDeltaRCM/debug_tools.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+class debug_tools(object):
+    """Debugging tools.
+
+    These tools can be invoked as part of a script during runtime or in a
+    Python debugging shell.
+
+    Add ``breakpoint()`` to any line in the code to enter the debugger and use
+    these tools interactively at that breakpoint. Note, for Python < 3.7 use
+    pdb.set_trace().
```
Comment (elbeejay) [inline pyDeltaRCM/debug_tools.py]: Want to stylize it as ``pdb.set_trace()`` to be consistent with ``breakpoint()``?

**Review by elbeejay — APPROVED** [review]
Pretty neat idea. This works for me - as a set of debugging tools I am not concerned about errors or funny plots when you try to `show_attribute` for unusual attributes (like 'seed' which gives an error, or 'iwalk' which sticks a the little iwalk 3x3 in the corner of the grid plot). 

Should be helpful for all future debugging efforts! :+1:

**Comment by elbeejay** [comment]
Maybe this is wishful thinking, but do you know if there is a way to pass the command `import matplotlib.pyplot as plt` to the debugger? Unfortunate that the rest of these tools work but you can't display the plot without typing out that line in the debugger itself.

**Comment by amoodie** [comment]
Fixed e.g., `iwalk` by making the routine determine the grid size from the attribute itself.
image

Added a colorbar to all plots:
image

This one is kind of weird, since a lot of these plots are categorical...I think you could do some educated guessing to make the colormaps categorical for certain attributes, but I figured I'd just leave it as "dumb"...

still working on a few things, and why the tests are failing...

**Comment by amoodie** [comment]
> do you know if there is a way to pass the command `import matplotlib.pyplot as plt` to the debugger? Unfortunate that the rest of these tools work but you can't display the plot without typing out that line in the debugger itself.

This is a good idea, but I could not find any way to do it. One option would be to add the import to the top of every module file. This is "bad practice" according to pep etc, but probably fine because we all know that `plt` is sort-of reserved for matplotlib. I'd be fine with adding this though. We could consider it in the future?

Otherwise, I got the tests to pass by removing the text from plots for testing (Windows has a different Freetype library used by matplotlib than *nix platforms. Also, we now explicitly create figures with specified size in the test suite --- for some reason a few figures were being created as different sizes on ubuntu without fixing the size.

Good to merge this, @elbeejay ?

**Pull request merged**